### PR TITLE
[matcher] [coordinator] Add RequireNamespaceWatchOnInit option

### PR DESCRIPTION
**Pull request by wesleyk**
This option on the matcher will require that namespace
values and corresponding rulesets are loaded
in the matcher as part of startup.

This is useful in ensuring dynamic rules are loaded
before we start ingesting metrics.

**What this PR does / why we need it**:

Fixes #

**Special notes for your reviewer**:

**Does this PR introduce a user-facing and/or backwards incompatible change?**:

```release-note

```

**Does this PR require updating code package or user-facing documentation?**:

```documentation-note

```

### CODE REVIEW

File: src/metrics/matcher/namespaces.go [hunk 1]
```diff
@@ -302,6 +315,13 @@ func (n *namespaces) process(value interface{}) error {
 					n.log.Error("failed to watch ruleset updates",
 						zap.String("ruleSetKey", ruleSet.Key()),
 						zap.Error(err))
+
+					// Track errors if we explicitly want to ensure watches succeed.
+					if n.requireNamespaceWatchOnInit {
```
Comment (wesleyk) [inline src/metrics/matcher/namespaces.go:320]: one awkward bit here is that this also covers the update case, not just the init case. However even if we propagate an error in that case, in the update path it gets swallowed regardless

Comment (robskillington) [reply src/metrics/matcher/namespaces.go:320]: yeah makes sense

**Review by robskillington — APPROVED** [review]
LGTM but can we make the coordinator by default set this to true?

**Comment by wesleyk** [comment]
Need to default this to false since values won't always be set

**Pull request merged**